src/routes/auth.go

```diff
@@ -0,0 +1,39 @@
+package routes
+
+import (
+	"github.com/dana-team/platform-backend/src/types"
+	"github.com/dana-team/platform-backend/src/utils"
+	"github.com/gin-gonic/gin"
+	"net/http"
+)
+
+// Login handles user authentication and issues a token on successful login.
+// It expects a JSON payload containing the user's credentials:
+//
+//	{
+//	   "username": "your_username", "password": "your_password"
+//	}
+//
+// It responds with a JSON object containing the token on successful login:
+//
+//	{
+//	  "token": "your_generated_token"
+//	}
+//
+// If there's an error during authentication, it responds with an appropriate error message.
+func Login() gin.HandlerFunc {
+	return func(c *gin.Context) {
+		var user types.User
+		if err := c.ShouldBindJSON(&user); err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": "Invalid request payload"})
+			return
+		}
+
+		token, err := utils.ObtainOpenshiftToken(user.Username, user.Password)
```

Comment:
Would you like to check if `user.Username` or `user.Password` is empty? Please check if it validates it.

Reply:
The function will return invalid credentials in this case, so I don't see a reason to check this. WDYT?

Reply:
I think that in general it's good idea to validate parameters before continuing execution of the function. As you wish, it can also work as you did 👍

Reply:
I agree that we should validate parameters before continuing execution of the function. With binding:\"required\" added to the Credentials properties, empty parameters will now fail with the error 'Invalid request payload' right at the beginning.

Reply:
Yep I see. thanks!